### PR TITLE
feat: add bank payout test case templates and implementation for Adye…

### PR DESCRIPTION
**Pull request by kashif-m**
## Type of Change

- [ ] Bugfix
- [ ] New feature
- [x] Enhancement
- [ ] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

### Additional Changes

- [ ] This PR modifies the API contract
- [ ] This PR modifies the database schema
- [ ] This PR modifies application configuration/environment variables

## How did you test it?
Local testing was done on below connectors
- adyen
- wise

**Adyen**
image

**Wise**
image

## Checklist

- [x] I formatted the code `cargo +nightly fmt --all`
- [x] I addressed lints thrown by `cargo clippy`
- [x] I reviewed submitted code
- [x] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable

### CODE REVIEW

**Comment by jagan-jaya** [comment]
Please attach the screenshot of the tests you ran in `how did you test?` section

File: connector-template/test.rs [hunk 1]
```diff
@@ -418,6 +421,215 @@ async fn should_fail_for_refund_amount_higher_than_payment_amount() {
...
+    let payment_info = PaymentInfo {
+        payout_method_data: Some(api::PayoutMethodData::Card(api::payouts::CardPayout {
+            card_number: CardNumber::from_str("4111111111111111").unwrap(),
+            expiry_month: Secret::new("3".to_string()),
+            expiry_year: Secret::new("2030".to_string()),
+            card_holder_name: Secret::new("John Doe".to_string()),
+        }))
+    };
+    let response = CONNECTOR
+        .fulfill_card_payout(payment_info)
```
Comment (jagan-jaya) [inline connector-template/test.rs:606]:
```suggestion
        .fulfill_payout(payment_info)
```

File: crates/router/tests/connectors/adyen.rs [hunk 2]
```diff
@@ -453,6 +460,127 @@ async fn should_fail_capture_for_invalid_payment() {
     );
 }
 
+/******************** Payouts test cases ********************/
+// Create SEPA payout
+#[cfg(feature = "payouts")]
+#[actix_web::test]
+async fn should_create_sepa_payout() {
+    let mut payment_info = AdyenTest::get_payment_info();
```
Comment (jagan-jaya) [inline crates/router/tests/connectors/adyen.rs:468]: add all your default data for payout within get_payout_info and use it in all other places below
```suggestion
    let mut payout_info = AdyenTest::get_payout_info();
```

File: crates/router/tests/connectors/adyen.rs [hunk 3]
```diff
@@ -453,6 +460,127 @@ async fn should_fail_capture_for_invalid_payment() {
...
+        enums::PayoutStatus::RequiresFulfillment
+    );
+}
+
+// Create and fulfill SEPA payout
+#[cfg(feature = "payouts")]
+#[actix_web::test]
+async fn should_create_and_fulfill_sepa_payout() {
+    let mut payment_info = AdyenTest::get_payment_info();
+    payment_info = payment_info.map(|mut pi| {
```
Comment (jagan-jaya) [inline crates/router/tests/connectors/adyen.rs:496]: add all your default data for payout within get_payout_info and use it in all other places below
```suggestion
    let mut payout_info = AdyenTest::get_payout_info();
```

File: crates/router/tests/connectors/adyen.rs [hunk 4]
```diff
@@ -453,6 +460,127 @@ async fn should_fail_capture_for_invalid_payment() {
...
+        .await
+        .expect("Payout bank creation and fulfill response");
+    assert_eq!(response.status.unwrap(), enums::PayoutStatus::Success);
+}
+
+// Verifies if card is eligible for payout
+#[cfg(feature = "payouts")]
+#[actix_web::test]
+async fn should_verify_payout_eligibility() {
+    let mut payment_info = AdyenTest::get_payment_info();
```
Comment (jagan-jaya) [inline crates/router/tests/connectors/adyen.rs:519]: add all your default data for payout within get_payout_info and use it in all other places below
```suggestion
    let mut payout_info = AdyenTest::get_payout_info();
```

File: crates/router/tests/connectors/adyen.rs [hunk 5]
```diff
@@ -453,6 +460,127 @@ async fn should_fail_capture_for_invalid_payment() {
...
+        response.status.unwrap(),
+        enums::PayoutStatus::RequiresCreation
+    );
+}
+
+// Fulfills card payout
+#[cfg(feature = "payouts")]
+#[actix_web::test]
+async fn should_fulfill_card_payout() {
+    let mut payment_info: Option<PaymentInfo> = AdyenTest::get_payment_info();
```
Comment (jagan-jaya) [inline crates/router/tests/connectors/adyen.rs:543]: add all your default data for payout within get_payout_info and use it in all other places below
```suggestion
    let mut payout_info = AdyenTest::get_payout_info();
```

File: crates/router/tests/connectors/adyen.rs [hunk 6]
```diff
@@ -453,6 +460,127 @@ async fn should_fail_capture_for_invalid_payment() {
...
+        .await
+        .expect("Payout fulfill response");
+    assert_eq!(response.status.unwrap(), enums::PayoutStatus::Success);
+}
+
+// Cancels a created bank payout
+#[cfg(feature = "payouts")]
+#[actix_web::test]
+async fn should_create_and_cancel_created_payout() {
+    let mut payment_info = AdyenTest::get_payment_info();
```
Comment (jagan-jaya) [inline crates/router/tests/connectors/adyen.rs:564]: add all your default data for payout within get_payout_info and use it
```suggestion
    let mut payout_info = AdyenTest::get_payout_info();
```

File: crates/router/tests/connectors/wise.rs [hunk 7]
```diff
@@ -1 +1,167 @@
...
+}
+
+static CONNECTOR: WiseTest = WiseTest {};
+
+/******************** Payouts test cases ********************/
+// Creates a recipient at connector's end
+#[cfg(feature = "payouts")]
+#[actix_web::test]
+async fn should_create_payout_recipient() {
+    let mut payment_info = WiseTest::get_payment_info();
```
Comment (jagan-jaya) [inline crates/router/tests/connectors/wise.rs:65]: add all your default data for payout within get_payout_info and use it in all other places below
```suggestion
    let mut payout_info = WiseTest::get_payout_info();
```

File: crates/router/tests/connectors/wise.rs [hunk 8]
```diff
@@ -1 +1,167 @@
...
+        create_res.status.unwrap(),
+        enums::PayoutStatus::RequiresFulfillment
+    );
+}
+
+// Create and fulfill BACS payout
+#[cfg(feature = "payouts")]
+#[actix_web::test]
+async fn should_create_and_fulfill_bacs_payout() {
+    let mut payment_info = WiseTest::get_payment_info();
```
Comment (jagan-jaya) [inline crates/router/tests/connectors/wise.rs:134]: add all your default data for payout within get_payout_info and use it in all other places below
```suggestion
    let mut payout_info = WiseTest::get_payout_info();
```

File: crates/router/tests/connectors/wise.rs [hunk 9]
```diff
@@ -1 +1,167 @@
...
+        response.status.unwrap(),
+        enums::PayoutStatus::RequiresCreation
+    );
+}
+
+// Create BACS payout
+#[cfg(feature = "payouts")]
+#[actix_web::test]
+async fn should_create_bacs_payout() {
+    let mut payment_info = WiseTest::get_payment_info();
```
Comment (jagan-jaya) [inline crates/router/tests/connectors/wise.rs:92]: add all your default data for payout within get_payout_info and use it in all other places below
```suggestion
    let mut payout_info = WiseTest::get_payout_info();
```

**Review by jagan-jaya — CHANGES_REQUESTED** [review]

**Comment by jagan-jaya** [comment]
@kashif-m Good job on the tests, Nicely done 👍

**Review by jagan-jaya — APPROVED** [review]

**Review by jarnura — APPROVED** [review]

**Pull request merged**